src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/FrameComponentStateManager.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Tizen.Applications.ComponentBased.Common
+{
+    internal class FrameComponentStateManager : ComponentStateManger
+    {
+        private Interop.CBApplication.FrameLifecycleCallbacks _callbacks;
+
+        internal FrameComponentStateManager(Type ctype, string id, ComponentBasedApplication parent) : base(ctype, id, parent)
+        {
+            _callbacks.OnAction = new Interop.CBApplication.FrameActionCallback(OnActionCallback);
+            _callbacks.OnDeviceOrientationChanged = new Interop.CBApplication.FrameDeviceOrientationChangedCallback(OnDeviceOrientationChangedCallback);
+            _callbacks.OnLanguageChanged = new Interop.CBApplication.FrameLanguageChangedCallback(OnLanguageChangedCallback);
+            _callbacks.OnLowBattery = new Interop.CBApplication.FrameLowBatteryCallback(OnLowBatteryCallback);
+            _callbacks.OnLowMemory = new Interop.CBApplication.FrameLowMemoryCallback(OnLowMemoryCallback);
+            _callbacks.OnRegionFormatChanged = new Interop.CBApplication.FrameRegionFormatChangedCallback(OnRegionFormatChangedCallback);
+            _callbacks.OnRestore = new Interop.CBApplication.FrameRestoreCallback(OnRestoreCallback);
+            _callbacks.OnSave = new Interop.CBApplication.FrameSaveCallback(OnSaveCallback);
+            _callbacks.OnSuspendedState = new Interop.CBApplication.FrameSuspendedStateCallback(OnSuspendedStateCallback);
+            _callbacks.OnCreate = new Interop.CBApplication.FrameCreateCallback(OnCreateCallback);
+            _callbacks.OnDestroy = new Interop.CBApplication.FrameDestroyCallback(OnDestroyCallback);
+            _callbacks.OnPause = new Interop.CBApplication.FramePauseCallback(OnPauseCallback);
+            _callbacks.OnResume = new Interop.CBApplication.FrameResumeCallback(OnResumeCallback);
+            _callbacks.OnStart = new Interop.CBApplication.FrameStartCallback(OnStartCallback);
+            _callbacks.OnStop = new Interop.CBApplication.FrameStopCallback(OnStopCallback);
+            Parent = parent;
+        }
+
+        private IntPtr OnCreateCallback(IntPtr context, IntPtr userData)
+        {
+            FrameComponent fc = Activator.CreateInstance(ComponentClassType) as FrameComponent;
+            if (fc == null)
+                return IntPtr.Zero;
+
+            string id;
+            Interop.CBApplication.GetInstanceId(context, out id);
+            fc.Bind(context, ComponentId, id, Parent);
+
+            IntPtr winHandle;
+            IWindowInfo win = fc.CreateWindowInfo();
+            if (win == null)
+                return IntPtr.Zero;
+
+            bool result = fc.OnCreate();
+            if (!result)
```

Comment:
result is only used on `if` so, we can remove it
```suggestion
            if (!fc.OnCreate())
```

Reply:
Done